test/apps/iodemo/io_demo.cc

```diff
@@ -1820,11 +1833,12 @@ class DemoClient : public P2pDemoCommon {
                 // after wait_for_responses(), if some clients were closed in
                 // UCP Worker progress during handling of remote disconnection
                 // from servers
+                progress();
```

Comment:
why need additional progress here?

Reply:
otherwise - in case of no active servers w/o `uncompelted==window` we do no progress at all

Reply:
btw, I think it is worth waiting for resources instead of just progressing.
just progressing can be a problem, we could do it endlessly, but some connections are not restored.